include/aws/cal/hash.h

```diff
@@ -41,6 +43,15 @@ AWS_CAL_API struct aws_hash *aws_sha256_new(struct aws_allocator *allocator);
  * Allocates and initializes an md5 hash instance.
  */
 AWS_CAL_API struct aws_hash *aws_md5_new(struct aws_allocator *allocator);
+/**
+ * Allocates and initializes a crc32 hash instance
+ */
+AWS_CAL_API struct aws_hash *aws_crc32_new(struct aws_allocator *allocator);
```

Comment:
Add a big old comment explaining HOW users should extract the 32bit number from the buffer passed to finalize. copy/paste comment to the other crc32c_new()